vanilla/addons
# Add ban by Fingerprint to the Troll management plugin

This is the 'plugin' part for the following issue: https://github.com/vanillaforums/mse/issues/236
As detailed by the following estimate: https://github.com/vanillaforums/estimates/blob/master/mse/mse-troll-plugin-enhancements.md

This is complementary to the following PR: https://github.com/vanilla/vanilla-cloud/pull/2303

**How to test**

- Enable the **Troll management plugin**.
- Configure the plugin.
- Enable fingerprint checks.
- Set a maximum allowed number of user accounts tied to a single fingerprint.
- Logout from the admin account.
- Create a few new user accounts to reach the maximum allowed number of user accounts tied to a single fingerprint.
- Logout & back again as your administrative account.
- Reach the applicants management interface(/dashboard/user/applicants). Your newest user should be listed.

**Notes**

- Added configurations & corresponding screens to allow modification of the amount accounts sharing the same fingerprint.
- A few tweaks were done to the _fingerprinting_ process.
- The user registration process verifies if the number of allowed accounts using the same fingerprint is reached. If it is the case, the newly registered user is set as _applicant_.
